hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/MergeOnReadIncrementalRelation.scala

```diff
@@ -181,25 +181,25 @@ trait HoodieIncrementalRelationTrait extends HoodieBaseRelation {
     listAffectedFilesForCommits(conf, new Path(metaClient.getBasePath), commitsMetadata)
   }
 
-  protected lazy val (shouldEqualStartTimestamp, filterStartTimestamp) = if (startInstantArchived) {
+  protected lazy val (includeStartTime, startTs) = if (startInstantArchived) {
     (false, startTimestamp)
   } else {
     (true, includedCommits.head.getTimestamp)
   }
-  protected lazy val filterEndTimestamp = if (endInstantArchived) endTimestamp else includedCommits.last.getTimestamp
+  protected lazy val endTs: String = if (endInstantArchived) endTimestamp else includedCommits.last.getTimestamp
 
   // Record filters making sure that only records w/in the requested bounds are being fetched as part of the
   // scan collected by this relation
   protected lazy val incrementalSpanRecordFilters: Seq[Filter] = {
     val isNotNullFilter = IsNotNull(HoodieRecord.COMMIT_TIME_METADATA_FIELD)
 
-    val largerThanFilter = if (shouldEqualStartTimestamp) {
-      GreaterThan(HoodieRecord.COMMIT_TIME_METADATA_FIELD, filterStartTimestamp)
+    val largerThanFilter = if (includeStartTime) {
```

Comment:
Do we still need to keep the filters?

Reply:
In the case of `spark.sql.parquet.recordLevelFilter.enabled=true`, it still helps to filter.

If we plan to enable `spark.sql.parquet.enableVectorizedReader=true` later, this filter can be removed.

Reply:
Can we add some test cases for it?